src/main/java/qna/domain/Answer.java

```diff
@@ -41,26 +42,36 @@ public Answer(User writer, Question question, String contents) {
     }
 
     public Answer(Long id, User writer, Question question, String contents) {
-        this.id = id;
+        this(id, contents, question, writer, false);
+    }
 
-        if (Objects.isNull(writer)) {
-            throw new UnAuthorizedException();
-        }
+    private Answer(Long id, String contents, Question question, User writer, boolean deleted) {
+        validateWriter(writer);
+        validateQuestion(question);
```

Comment:
제가 평소 실무에 정의하는 순서를 그대로 적용해서 이렇게 구현하게 되었습니다.

이 순서를 사용했던 이유는 다른 개발자들이 클래스를 읽을 때 `public메소드`만 보고도 어떤일을 하는지 실제 구현을 다 보지 않고도 파악이 가능해야한다고 생각했기 때문입니다.

하지만 이슬님께서 제안하신 방법도 좋은 방법이고 가독성을 올리는데 충분히 도움을 줄 수 있을것  같습니다.
이와 관련해 찾아봤는데 오라클에서 제공하는 [Code Conventions for the Java Programming Language](https://www.oracle.com/java/technologies/javase/codeconventions-fileorganization.html#1852)에서도 말씀하신것과 같이 

> These methods should be grouped by functionality rather than by scope or accessibility. For example, a private class method can be in between two public instance methods. The goal is to make reading and understanding the code easier.

**메소드 순서는 기능별로 그루핑하면 코드를 읽기 쉽게 이해하기 쉽게 만들수 있다고 합니다.**

제안해 주신 방법으로 리팩토링해서 다시 리뷰요청 하도록 하겠습니다!

감사합니다 👍